src/observer.ts

```diff
@@ -40,14 +44,31 @@ class PerformanceObserver implements PollingPerformanceObserver {
   }
 
   public observe(options?: PerformanceObserverInit): void {
-    if (!options || !options.entryTypes) {
+    if (!options || (!options.entryTypes && !options.type)) {
       throw new Error(ERRORS["no-entry-types"]);
     }
 
-    const entryTypes: string[] = options.entryTypes.filter(isValidType);
+    if (options.entryTypes && options.type) {
+      throw new Error(ERRORS["both-entry-types"]);
+    }
+
+    let rawEntryTypes: string[] = [];
+    if (options && options.entryTypes) {
```

Comment:
Redundant check for `if (options)`: `options` cannot be undefined or null at this point. Could be simplified to `if (options.entryTypes)`.